include/seqan3/io/alignment_file/detail.hpp

```diff
@@ -227,4 +231,189 @@ std::string get_cigar_string(alignment_type && alignment,
                             query_start_pos, query_end_pos, extended_cigar);
 }
 
+/*!\brief Transforms a cigar string into a vector of operation-count pairs (e.g. (M, 3)).
+ * \ingroup alignment_file
+ *
+ * \tparam sc_begin_count_type  The type of the sc_begin_count value; must model std::Integral.
+ * \tparam sc_end_count_type    The type of the sc_end_count value; must model std::Integral.
+ * \tparam cigar_view_type      The type of a single pass input view over the cigar string; must model
+ *                              std::ranges::InputRange.
+ *
+ * \param[out] sc_begin_count  Will be set to the number of soft clipped bases at the beginning of the alignment.
+ * \param[out] sc_end_count    Will be set to the number of soft clipped bases at the end of the alignment.
+ * \param[in]  cigar_view      The single pass input view over the cigar string to parse.
+ *
+ * \returns A tuple of size three containing (1) std::vector of operation-count pairs, e.g. (M, 3), that describe
+ *          the alignment, (2) the aligned reference length and (3) the aligned sequence length.
+ *
+ * \details
+ *
+ * For example, the view over the cigar string "1S4M1D2M2S" will set sc_begin_count to 1, sc_end_count to 2
+ * and return `{[(M,4), (D,1), (M,2)], 7, 6}`.
+ */
+template <std::Integral sc_begin_count_type,
+          std::Integral sc_end_count_type,
+          std::ranges::InputRange cigar_view_type>
+std::tuple<std::vector<std::pair<char, size_t>>, size_t, size_t> parse_cigar(sc_begin_count_type & sc_begin_count,
+                                                                             sc_end_count_type & sc_end_count,
+                                                                             cigar_view_type && cigar_view)
```

Comment:
Because I need the generic types to avoid type conversion. :/

Reply:
But there is no conversion when you bind lvalue references...

Reply:
Is this only used in detail?

Mayber just use a struct?
```cpp
struct cigar
{
    std::vector<size_t> lengths;
    std::string chars;
    std::pair<size_t, size_t> soft_clips;
};
```
?

Reply:
yes maybe..

Reply:
hmm on second thought, no it wouldn't because I could not use 
`std::tie(par1, par2, ..) = parse_cigar(...)` otherwise but would need to assign 5 variables after returning the struct.